test.py

```diff
@@ -48,6 +48,13 @@ def test_opaque(self):
         assert_array_equal(pos, np.arange(10))
         assert_array_equal(neg, -np.arange(10))
 
+    def test_store(self):
+        stored = self.fut.test10a(42)
+        intermediate = self.fut.store_testOpaque(stored)
+        restored = self.fut.restore_testOpaque(intermediate)
+        self.assertEqual(self.fut.test10b(stored), self.fut.test10b(restored))
+
+
```

Comment:
Could you also add a negative test where you pass in some garbage to restore?

Reply:
How would that garbage look like? In this case the value is just a scalar, so without knowing the exact representation, I think most byte sequences could probably be something valid unless there is some kind of hash built in.

Reply:
There is a header and type-based metadata.  But you can just pass an empty byte sequence.